From 19d6f4237bb2c4318d3ec0921c55ecd8a3c02361 Mon Sep 17 00:00:00 2001
From: Chen Linxuan <me@black-desk.cn>
Date: Fri, 5 Jun 2026 09:47:52 +0800
Subject: [PATCH] Declare transitive deps in rime.pc for static linking

Signed-off-by: Chen Linxuan <me@black-desk.cn>
---
 CMakeLists.txt | 5 +++++
 rime.pc.in     | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c170adfe62a..03d66a216791 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -212,6 +212,11 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux|FreeBSD|OpenBSD|DragonFly|GNU|Darwin" OR
   set(pkgdatadir "${CMAKE_INSTALL_FULL_DATADIR}/${RIME_DATA_DIR}")
   set(pluginsdir "${CMAKE_INSTALL_FULL_LIBDIR}/${RIME_PLUGINS_DIR}")
   set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
+  if(APPLE)
+    set(RIME_CXX_LINK "-lc++")
+  else()
+    set(RIME_CXX_LINK "-lstdc++")
+  endif()
   configure_file(
       ${PROJECT_SOURCE_DIR}/rime.pc.in
       ${PROJECT_BINARY_DIR}/rime.pc
diff --git a/rime.pc.in b/rime.pc.in
index 1a8c54318398..de0097f68811 100644
--- a/rime.pc.in
+++ b/rime.pc.in
@@ -10,3 +10,5 @@ Description: Rime Input Method Engine
 Version: @rime_version@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lrime
+Requires.private: gflags, marisa, opencc, yaml-cpp
+Libs.private: @RIME_CXX_LINK@ -lboost_filesystem -lboost_locale -lboost_thread -lboost_chrono -lboost_atomic -lboost_container -lboost_date_time -lboost_charconv -ldarts -lleveldb -llua -lglog
-- 
2.53.0

